packages/next/src/server/stream-utils/node-web-streams-helper.ts

```diff
@@ -648,17 +704,26 @@ export async function continueDynamicHTMLResume(
     inlinedDataStream,
     getServerInsertedHTML,
     getServerInsertedMetadata,
+    nonce,
   }: ContinueResumeOptions
 ) {
   return (
     renderStream
       // Buffer everything to avoid flushing too frequently
       .pipeThrough(createBufferedTransformStream())
       // Insert generated tags to head
-      .pipeThrough(createHeadInsertionTransformStream(getServerInsertedHTML))
+      .pipeThrough(
+        createHeadInsertionTransformStream(getServerInsertedHTML, {
+          nonce,
+          reinsertIcons: false,
+        })
+      )
       // Insert generated metadata to body
       .pipeThrough(
-        createHeadInsertionTransformStream(getServerInsertedMetadata)
+        createHeadInsertionTransformStream(getServerInsertedMetadata, {
+          nonce,
+          reinsertIcons: true,
+        })
```

Comment:
why can't `getServerInsertedMetadata` just return the script as part of the html it produces. I wasn't expecting that any new config here in the stream handler would actually be necessary. Ideally it's a black box and the fact we have this insertion instruction is just part of what the server inserted metadata chooses to encode